bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -673,65 +712,44 @@ protected synchronized RuntimeRule getRuntimeRule(String rUID) {
      * @param rUID unique id of the rule
      * @param isEnabled true to enable the rule, false to disable it
      */
-    protected void setRuleEnabled(Rule rule, boolean isEnabled) {
-        String rUID = rule.getUID();
-        RuleStatus status = getRuleStatus(rUID);
-        String enabled = isEnabled ? "enabled" : "disabled";
+    protected void setRuleEnabled(String rUID, boolean isEnabled) {
         RuntimeRule runtimeRule = getRuntimeRule(rUID);
         if (runtimeRule == null) {
-            logger.debug("There is no rule with UID '{}' which could be {}", rUID, enabled);
             return;
         }
+        RuleStatusInfo statusInfo = runtimeRule.getStatusInfo();
         if (isEnabled) {
-            if (status == RuleStatus.DISABLED) {
-                setRule(runtimeRule);
-            } else {
-                logger.debug("The rule rId = {} is already enabled.", rUID);
-            }
-        } else {
-            unregister(runtimeRule);
-            setRuleStatusInfo(rUID, new RuleStatusInfo(RuleStatus.DISABLED), true);
-        }
-    }
-
-    private void addNewModuleTypes(ModuleHandlerFactory mhf, Collection<String> moduleTypes) {
-        Set<String> notInitailizedRules = null;
-        for (Iterator<String> it = moduleTypes.iterator(); it.hasNext();) {
-            String moduleTypeName = it.next();
-            Set<String> rules = null;
-            synchronized (this) {
-                moduleHandlerFactories.put(moduleTypeName, mhf);
-                Set<String> rulesPerModule = mapModuleTypeToRules.get(moduleTypeName);
-                if (rulesPerModule != null) {
-                    rules = new HashSet<String>();
-                    rules.addAll(rulesPerModule);
-                }
-            }
-            if (rules != null) {
-                for (String rUID : rules) {
-                    RuleStatus ruleStatus = getRuleStatus(rUID);
-                    if (ruleStatus == RuleStatus.UNINITIALIZED) {
-                        notInitailizedRules = notInitailizedRules != null ? notInitailizedRules
-                                : new HashSet<String>(20);
-                        notInitailizedRules.add(rUID);
-                    }
-
+            if (statusInfo.getStatus().equals(RuleStatus.DISABLED)) {
+                if (statusInfo.getStatusDetail().equals(RuleStatusDetail.NONE)
```

Comment:
This looks like a piece of code that will definitely be forgotten to be updated the next time somebody adds a new status detail... What is this exactly filtering for here? Can this be expressed differently?

Reply:
I'm checking if the status detail can be changed by the RuleEngine. The other status details are managed by RuleRegistry, so there is no need to call `setRule(runtimeRule)` to try to initialize the rule. I guess this check can be moved in a separate method - something like `isManagedByRuleRegistry()` in RuleStatusDetail?

Reply:
As mentioned above, the RuleRegistry imho must not manage any status.